apps/avatax/e2e/tests/cheapest_voucher_product_without_tax_class.spec.ts

```diff
@@ -0,0 +1,214 @@
+import { e2e } from "pactum";
+import { describe, it } from "vitest";
+import {
+  CheckoutAddVoucher,
+  CheckoutUpdateDeliveryMethod,
+  CompleteCheckout,
+  CreateCheckoutWithTwoLines,
+  MoneyFragment,
+} from "../generated/graphql";
+import { string } from "pactum-matchers";
+
+describe("App should calculate taxes for checkout with cheapest voucher applied [pricedEnteredWithTax: False]", () => {
+  const testCase = e2e(
+    "Product without tax class [pricesEnteredWithTax: False], cheapest voucher applied",
+  );
+
+  const CURRENCY = "USD";
+
+  const TOTAL_TAX_PRICE_BEFORE_SHIPPING = 41.7;
+  const TOTAL_NET_PRICE_BEFORE_SHIPPING = 469.9;
+  const TOTAL_GROSS_PRICE_BEFORE_SHIPPING = 511.6;
+
+  const TOTAL_NET_PRICE_AFTER_SHIPPING = 565.23;
+  const TOTAL_TAX_PRICE_AFTER_SHIPPING = 50.16;
+  const TOTAL_GROSS_PRICE_AFTER_SHIPPING = 615.39;
+
+  const SHIPPING_NET_PRICE = 95.33;
+  const SHIPPING_TAX_PRICE = 8.45;
+  const SHIPPING_GROSS_PRICE = 103.78;
+
+  const PRODUCT_ONE_NET_PRICE_BEFORE_VOUCHER = 19.9;
+  const PRODUCT_ONE_TAX_PRICE_BEFORE_VOUCHER = 1.77;
+  const PRODUCT_ONE_GROSS_PRICE_BEFORE_VOUCHER = 21.67;
+
+  const PRODUCT_ONE_NET_PRICE_AFTER_VOUCHER = 19.5;
+  const PRODUCT_ONE_TAX_PRICE_AFTER_VOUCHER = 1.73;
+  const PRODUCT_ONE_GROSS_PRICE_AFTER_VOUCHER = 21.23;
+
+  const VOUCHER_AMOUNT = 0.4;
+
+  const PRODUCT_TWO_NET_PRICE_BEFORE_VOUCHER = 450;
+  const PRODUCT_TWO_TAX_PRICE_BEFORE_VOUCHER = 39.94;
+  const PRODUCT_TWO_GROSS_PRICE_BEFORE_VOUCHER = 489.94;
+
+  const PRODUCT_TWO_NET_PRICE_AFTER_VOUCHER = PRODUCT_TWO_NET_PRICE_BEFORE_VOUCHER;
+  const PRODUCT_TWO_TAX_PRICE_AFTER_VOUCHER = PRODUCT_TWO_TAX_PRICE_BEFORE_VOUCHER;
+  const PRODUCT_TWO_GROSS_PRICE_AFTER_VOUCHER = PRODUCT_TWO_GROSS_PRICE_BEFORE_VOUCHER;
+
+  const TOTAL_NET_PRICE_AFTER_VOUCHER = 564.83;
+  const TOTAL_TAX_PRICE_AFTER_VOUCHER = 50.13;
+  const TOTAL_GROSS_PRICE_AFTER_VOUCHER = 614.96;
+
+  const getMoney = (amount: number): MoneyFragment => {
+    return {
+      amount,
+      currency: CURRENCY,
+    };
+  };
+
+  const getCompleteMoney = ({ gross, net, tax }: { gross: number; net: number; tax: number }) => ({
```

Comment:
```suggestion
  const getTaxedMoney = ({ gross, net, tax }: { gross: number; net: number; tax: number }) => ({
```

**Suggestion:** In GraphQL schema this is called `TaxedMoney` maybe we can add it here as well?